Automattic/wp-calypso
# Migrate Domain Management CSS styles to webpack

Migrate Domains CSS styles to webpack. Splits the big stylesheets into smaller ones and binds them to specific React components. Ensures that the three sections that use the Domains components:
- Checkout: domain contact details
- Domain Management: adding and managing domains and their DNS records
- Domain Connect Authorize: endpoint to automatically add DNS records for services like GSuite or Office365
get exactly the styles they need. Nothing missing, nothing redundant.

The more complicated parts are:
- splitting the `client/my-sites/domains/domain-management/style.scss` stylesheet into individual pieces
- splitting the `*__explanation` styles from one rule to multiple stylesheets (would deserve its own component in the future)
- creating reusable components for `DnsRecordsList` and `DnsRecordsListItem` instead of copying raw `<ul class="dns__list">` markup.

**How to test:**
Test the following parts of UI:
- checkout: domain contact details
- domains list: setting primary domain, adding new domain
- domain details, contact info, privacy
- transfers: in, out, some more obscure bits like IPS codes for .uk domains or site redirects
- DNS records, custom nameservers, setting them up from templates (GSuite, O365, Zoho)
- site redirects and domain-only sites
- Domain Connect: I use the following testing link to see the UI:
http://calypso.localhost:3000/domain-connect/authorize/v2/domainTemplates/providers/a8cdemo.blog/services/dc-test/apply?domain=example.blog?blerg=a&blerg-data=b&blarg=c&blarg-data=d
